pkg/controllers/vdb/verticadb_controller.go

```diff
@@ -96,7 +97,7 @@ func (r *VerticaDBReconciler) SetupWithManager(mgr ctrl.Manager, options control
 		Owns(&appsv1.Deployment{})
 
 	discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(mgr.GetConfig())
-	if isServiceMonitorObjectInstalled(discoveryClient) {
+	if opcfg.IsPrometheusEnabled() && isServiceMonitorObjectInstalled(discoveryClient) {
```

Comment:
Is this related to removing the service monitor? Should we add a test for this?

Reply:
This is related to an issue Abhishek found. Without this change, if prometheus is disabled and the servicemonitor CRD is installed in the cluster, the operator will try to own the servicemonitor, but because it does not have permissions (we add servicemonitor rule in the clusterrole only when prometheus is enabled) it will crash.

We do not need to add a test. Abhishek will do the testing.